src/_backend_agg.cpp

```diff
@@ -212,8 +212,8 @@ agg::rect_i RendererAgg::get_content_extents()
 
     r.x1 = std::max(0, r.x1 - 1);
     r.y1 = std::max(0, r.y1 - 1);
-    r.x2 = std::max(r.x2 + 1, (int)width);
-    r.y2 = std::max(r.y2 + 1, (int)height);
+    r.x2 = std::min(r.x2 + 1, (int)width);
+    r.y2 = std::min(r.y2 + 1, (int)height);
```

Comment:
It used to be called `tostring_rgba_minimized`.  And actually it looks like it used to have this fix, which got transferred incorrectly in the C++ refactor:

```
       // Expand the bounds by 1 pixel on all sides
        xmin = std::max(0, xmin - 1);
        ymin = std::max(0, ymin - 1);
        xmax = std::min(xmax, (int)width);
        ymax = std::min(ymax, (int)height);
```